main/src/cgeo/geocaching/activity/OAuthAuthorizationActivity.java

```diff
@@ -181,12 +183,15 @@ public void onResume() {
         final Uri uri = getIntent().getData();
         if (uri != null) {
             final String verifier = uri.getQueryParameter("oauth_verifier");
-            if (StringUtils.isNotBlank(verifier)) {
-                exchangeTokens(verifier);
-            } else {
-                // We can shortcut the whole verification process if we do not have a token at all.
-                changeTokensHandler.sendEmptyMessage(NOT_AUTHENTICATED);
+            if (!verifier.equals(lastVerifier)) {
+                if (StringUtils.isNotBlank(verifier)) {
+                    exchangeTokens(verifier);
+                } else {
+                    // We can shortcut the whole verification process if we do not have a token at all.
+                    changeTokensHandler.sendEmptyMessage(NOT_AUTHENTICATED);
+                }
             }
+            lastVerifier = verifier;
```

Comment:
Can you move this line into the `if` block?

Reply:
restructured to simplify the whole structure